main/src/cgeo/geocaching/location/DistanceParser.java

```diff
@@ -35,21 +34,70 @@ public static float parseDistance(final String distanceText, final boolean metri
         }
 
         final float value = Float.parseFloat(matcher.group(1).replace(',', '.'));
-        final String unit = StringUtils.lowerCase(matcher.group(2), Locale.US);
+        final String unitStr = StringUtils.lowerCase(matcher.group(2), Locale.US);
 
-        if ("m".equals(unit) || (StringUtils.isEmpty(unit) && metricUnit)) {
-            return value / 1000;
+        UNIT unit = metricUnit ? UNIT.M : UNIT.FT;
+
+        switch (unitStr) {
+            case "m":
+                unit = UNIT.M;
+                break;
+            case "km":
+                unit = UNIT.KM;
+                break;
+            case "yd":
+                unit = UNIT.YD;
+                break;
+            case "mi":
+                unit = UNIT.MI;
+                break;
+            case "ft":
+                unit = UNIT.FT;
+                break;
         }
-        if ("km".equals(unit)) {
-            return value;
+        return convertDistance(value, unit);
+    }
+
+    /**
+     * Converts distance from different units to kilometers
+     *
+     * @param distance source distance to convert
+     * @param unit     unit to convert from
+     * @return the distance in kilometers
+     */
+    public static float convertDistance(final float distance, final UNIT unit)
```

Comment:
For me "parse" is related to string parsing. That's why method reading string and extracting values from it is "parse...", but method accepting numbers is "convert" - because there is nothing to parse here. About the direction of conversion - that's why I put some Javadoc here.

Reply:
What I have in mind is one method `public static float parseDistance(final String distanceText, final DistanceUnit unit)`.

Reply:
done